packages/machine-extractor/src/extractAction.ts

```diff
@@ -377,31 +321,27 @@ function extractEventObject(
             isTemplateLiteralWithExpressions(prop.value) ||
             t.isNullLiteral(prop.value)
           ) {
-            extracted[prop.key.name] = {
-              type: 'expression',
-              value: fileContent.slice(prop.value.start!, prop.value.end!),
-            };
+            extracted[prop.key.name] = fileContent.slice(
+              prop.value.start!,
+              prop.value.end!,
+            );
           } else if (t.isTemplateLiteral(prop.value)) {
-            extracted[prop.key.name] = {
-              type: 'string',
-              value:
-                prop.value.quasis[0].value.cooked ??
-                prop.value.quasis[0].value.raw,
-            };
+            extracted[prop.key.name] =
+              prop.value.quasis[0].value.cooked ??
+              prop.value.quasis[0].value.raw;
           } else {
-            extracted[prop.key.name] = {
-              type: getLiteralType(prop.value),
-              value: prop.value.value,
-            } as Extract<JsonEntry, { type: 'string' | 'number' | 'boolean' }>;
+            extracted[prop.key.name] = prop.value.value;
           }
-        } else if (
-          t.isArrayExpression(prop.value) ||
-          t.isObjectExpression(prop.value)
-        ) {
-          extracted[prop.key.name] = {
-            type: 'expression',
-            value: fileContent.slice(prop.value.start!, prop.value.end!),
-          };
+        } else if (t.isArrayExpression(prop.value)) {
+          extracted[prop.key.name] = extractArrayRecursively(
+            prop.value,
+            fileContent,
+          );
+        } else if (t.isObjectExpression(prop.value)) {
+          extracted[prop.key.name] = extractObjectRecursively(
+            prop.value,
+            fileContent,
+          );
         } else {
           console.warn(
```

Comment:
This is IMHO a little bit problematic because it favors partial results. A better approach would be to extract everything that contains unsupported things as inline. I understand though that you might not want to change this rn.

Reply:
What key value pair would not be extracted here? I want to understand the partial part.